Plugins/Creature/Creature.cpp

```diff
@@ -1739,4 +1741,60 @@ ArgumentStack Creature::GetTotalEffectBonus(ArgumentStack&& args)
     Services::Events::InsertArgument(stack, retVal);
     return stack;
 }
+
+ArgumentStack Creature::SetOriginalName(ArgumentStack&& args)
+{
+    ArgumentStack stack;
+
+    if (auto *pCreature = creature(args))
+    {
+        const auto name = Services::Events::ExtractArgument<std::string>(args);
+          ASSERT_OR_THROW(!name.empty());
```

Comment:
Setting the last name to `""` should be okay.

Reply:
Moved the assert to firstname only